File: LeetCode/JavaScript/#367 Valid Perfect Square.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function(num) {
    if(num===1) return true;
    var high = Math.ceil(num/2);
    var low = 2;
    var mid;
    while(low<=high){
      mid = Math.ceil((low+high)/2);
      if(mid*mid<num){
        low = mid+1;
      }else if(mid*mid>num){
        high = mid-1;
      }else{
        break;
      }
    }
    return mid*mid===num;
};

File: LeetCode/Python/#367 Valid Perfect Square.py
import math
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num == 1:
            return True
        h = math.ceil(num/2)
        l = 2
        while l <= h:
            m = math.ceil((l+h)/2)
            if m*m < num:
                l = m + 1
            elif m*m>num:
                h = m - 1
            else:
                break

        return m*m == num
